src/base/bittorrent/trackerentry.cpp

```diff
@@ -52,6 +53,30 @@ QString TrackerEntry::url() const
     return QString::fromStdString(nativeEntry().url);
 }
 
+QStringList TrackerEntry::messages() const
+{
+    QStringList messages;
+    const auto addMessage = [&messages](const std::string &msg) {
+        QString message = QString::fromStdString(msg).trimmed();
+        if (!message.isEmpty() && !messages.contains(message))
+            messages << std::move(message);
+    };
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    addMessage(nativeEntry().message);
+    if (messages.isEmpty() && status() == TrackerEntry::NotWorking)
+        addMessage(nativeEntry().last_error.message());
+#else
+    const auto &endpoints = nativeEntry().endpoints;
```

Comment:
I would prefer to see the actual type here.

Reply:
I don't care about the type here, let the compiler do the job, we need some ranged collection with `announce_endpoint`. I changed to explicit types in loops so it is clear what type we are working on.